tests/websockets/src/jvmTest/kotlin/WebSocketErrorsTest.kt

```diff
@@ -216,4 +217,44 @@ class WebSocketErrorsTest {
 
     apolloClient.dispose()
   }
+
+  @Test
+  fun closeConnectionReconnectsTheWebSocket() = runBlocking {
+    class MyWebSocketReconnectException : Exception()
+
+    var connectionInitCount = 0
+    val apolloClient = ApolloClient.Builder()
+        .httpServerUrl("http://localhost:8080/graphql")
+        .webSocketServerUrl("http://localhost:8080/subscriptions")
+        .wsProtocol(
+            SubscriptionWsProtocol.Factory(
+                connectionPayload = {
+                  connectionInitCount++
+                  mapOf("return" to "success")
+                }
+            )
+        )
+        .webSocketReopenWhen { e, _ ->
+          assertIs<MyWebSocketReconnectException>(e)
+          true
+        }
+        .build()
+
+    apolloClient.subscription(CountSubscription(2, 100))
+        .toFlow()
+        .test {
+          awaitItem() // 0
+
+          (apolloClient.subscriptionNetworkTransport as WebSocketNetworkTransport).closeConnection(MyWebSocketReconnectException())
```

Comment:
Not great - we could have an extension on `NetworkTransport` that checks the type to make it a bit nicer to use?

Reply:
Agreed

```kotlin
fun NetworkTransport.closeConnection()
```

?